### ISSUE TITLE
Page Render Cache-Control

### ISSUE THREAD

**Issue by cdowdy** [bolt/bolt#4600]
1) __Bug Summary__.  
If you set caching for GET requests to true the page you are on/visiting doesn't get served from the cache if you visit/return within the duration set in the config.  

2) __Specifics__.  
* Bolt Version:  
  * All versions of Bolt.  
  * installed through composer, git, and download from site.  
* Browsers affected... All modern ( Edge, Webkit, Blink Based browsers and Firefox )  

3) __Steps to reproduce__:  

set caching for GET requests to true. 
```yaml  
caching:
    config: true
    templates: true
    request: true
    duration: 10
    authenticated: false  
```  
visit the page. Here's a good ol gif of that in process.    
  
![s-maxage-chrome](https://cloud.githubusercontent.com/assets/5969974/11847677/97e23eb6-a3ec-11e5-9d21-3d1718a4c1a2.gif)  

4)__Expected Result__  
page should be served from the cache.  

------------------------  
The browser looks for the "max-age" in the cache control header. The s-maxage is used for shared caches.  

Adding in a max-age cache-control directive will fix this. Here is another gif of that in process with a max age and s-maxage cache-control directives.  
![chrome-max-age](https://cloud.githubusercontent.com/assets/5969974/11848068/90356c40-a3ee-11e5-858b-2d6b9eb361cc.gif)  

As you can see with the max-age directive when you return to the previously rendered/visited page within the duration set in the caching config settings the page is served from the cache.  

Since you all are now using symfony's response and setcache methods this might also open up areas for extensions to either create a static page and serve those instead of calls to the db to render or serve the page with a 304 similar to http://symfony.com/doc/current/book/http_cache.html#optimizing-cache-validation  

 
I've gone ahead and created a pull request to "fix" this ( #4599 ) . If it's not what you are looking for then I'll just use it in my own deploys.  

Thanks!

**Comment by SahAssar** [comment]
Isn't setting that kind of header usually the job of the server config, not the app? The caching strictly refers to caching the rendered page on-server in bolt's own cache, not client-side caching. Sorry if I misunderstood :)

**Comment by cdowdy** [comment]
The app sets it. So even if you try to over ride it it gets set to ```s-maxage``` because it's called in render.php

**Comment by SahAssar** [comment]
Fixed in #4599, Thanks!
